### ISSUE TITLE
📜 make a "free play" adventure

### ISSUE THREAD

**Issue by Felienne** [hedyorg/hedy#6042]
Now that #6039 removes the developer mode, it would be lovely to add one "free play" adventure where kids can do whatever they want. The POs might want to think of a good layout for that :)

**Comment by AnneliesVlaar** [comment]
The free play adventure doesn't have any instruction text which gives space for an elongated input and output screen. You should be able to open programs, create new files and give a custom name to a file. 

I can imagine that when you're trying to create something you need to 'level up' to make things possible. Therefore I suggest a dropdown to easily switch between levels, to run the code in a different level. It might also be confusing when you don't know how to do things in a level (syntax and available concepts) So with the information button next to the level you can get a modal with some explanation.

![Image](https://github.com/user-attachments/assets/3e5cf350-a612-4f12-bc35-11fe7062bf4a)


To make it easier to learn/copy from other files we could open multiple files in a tab with the cross to close the tab. Would it then be possible to import files and functions from other files?
![Image](https://github.com/user-attachments/assets/9e6b46d9-b170-412b-85a2-4a0c6f899279)


When opening a new file, we should somehow indicate in which level this file is created/run. Here this is indicated between brackets (level), and the files are sorted alphabetically 

![Image](https://github.com/user-attachments/assets/b806cf4e-f6ff-4691-b07d-bcda2288f6b3)

**Comment by MarleenGilsing** [comment]
Oh wow! I thought we were simply going for an adventure with no explanation and a forced developers mode, but this is way more intricate. I do like it, but since we have limited resources on our programmers' time, maybe we can start out with a simple 'free play' adventure, and keep this on our low prio wishlist?

**Comment by AnneliesVlaar** [comment]
Yes once I start thinking about it, al kinds of problems and solutions came to my mind. 

But how do we expect the free play adventure will be used? We can already add an adventure with only the instruction 'do whatever you like'. Do we want extra features (like more lines and longer run time) to have an adventure 'free of limitations'?

**Comment by AnneliesVlaar** [comment]
> we can start out with a simple 'free play' adventure
Not sure if we need to force developers mode on the adventure. We can also limit the instructions, maybe like this?

![Image](https://github.com/user-attachments/assets/e1c3ae7f-5a1c-49a5-ba55-97740a187c83)

**Comment by MarleenGilsing** [comment]
I thought we could force the developers' mode, so the user feels they have a lot of space to create something (and since we are removing the developers mode for the other adventures it might feel more special)

**Comment by AnneliesVlaar** [comment]
More space, limited instructions:
![Image](https://github.com/user-attachments/assets/eeeab883-64e5-4057-90bc-b75f1e52336f)

And how about the maximum of 100 lines and the limited run time (not able to repeat many times see #5248)?

**Comment by MarleenGilsing** [comment]
agree we should disable those limitations

**Comment by MarleenGilsing** [comment]
@AnneliesVlaar is thinking about how to make this a doable issue for the hackathon and post it here! :)

**Comment by MarleenGilsing** [comment]
Love it! :D 

Let's maybe just call the adventure 'free play' though, students will instantly know they are allowed to play around freely. Sandbox might be confusing.

**Comment by delenopschool** [comment]
I love the idea!
I also really like the extend to 200 line option, but my program is really big, like 666 lines. So i was wandering, why is there even a limit on how much lines you can run?

**Comment by AnneliesVlaar** [comment]
Get insprition from https://github.com/hedyorg/hedy/pull/6091
